### PR TITLE
[Merged by Bors] - chore: add @[ext] lemmas for NatTrans synonyms

### PR DESCRIPTION
**Pull request by kim-em**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/MorphismProperty.lean [hunk 1]
```diff
@@ -494,6 +494,13 @@ lemma FunctorsInverting.ext {W : MorphismProperty C} {F₁ F₂ : FunctorsInvert
 instance (W : MorphismProperty C) (D : Type _) [Category D] : Category (FunctorsInverting W D) :=
   FullSubcategory.category _
 
+-- Porting note: add another `@[ext]` lemma since `ext` can't see through the definition to use
+-- `NatTrans.ext`.
+@[ext]
+lemma FunctorsInverting.hom_ext {W : MorphismProperty C} {F₁ F₂ : FunctorsInverting W D}
+    {α β : F₁ ⟶ F₂} (h : ∀ X : C, α.app X = β.app X) : α = β :=
+  NatTrans.ext _ _ (funext h)
```
Comment (eric-wieser) [inline Mathlib/CategoryTheory/MorphismProperty.lean:502]: Does
```suggestion
@[ext]
lemma FunctorsInverting.hom_ext {W : MorphismProperty C} {F₁ F₂ : FunctorsInverting W D}
    {α β : F₁ ⟶ F₂} (h : α.app = β.app) : α = β :=
  NatTrans.ext _ _ h
```
still work? Or does `h` not have the right type?

Comment (semorrison) [reply Mathlib/CategoryTheory/MorphismProperty.lean:503]: Yes, that's fine too. I've just never ever wanted these partially applied ext lemmas, and if I'm writing them by hand I'd prefer the useful one.

Comment (eric-wieser) [reply Mathlib/CategoryTheory/MorphismProperty.lean]: Isn't it the case that as long as you use the `ext` tactic, the partially-applied version is never less useful than the fully applied one?

I have no idea whether the usual argument *for* partially-applied lemmas is relevant here, as I don't know the type of α.app.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**